bindings/rust/s2n-tls/src/enums.rs

```diff
@@ -31,6 +31,29 @@ impl<T, E> From<Result<T, E>> for CallbackResult {
     }
 }
 
+#[non_exhaustive]
+#[derive(Debug, PartialEq, Copy, Clone)]
+pub enum FipsMode {
+    Disabled,
+    Enabled,
+}
+
+impl FipsMode {
+    pub fn is_enabled(&self) -> bool {
+        matches!(self, FipsMode::Enabled)
+    }
+}
+
+impl From<s2n_fips_mode::Type> for FipsMode {
+    fn from(input: s2n_fips_mode::Type) -> Self {
+        match input {
+            s2n_fips_mode::FIPS_MODE_DISABLED => FipsMode::Disabled,
+            s2n_fips_mode::FIPS_MODE_ENABLED => FipsMode::Enabled,
+            _ => unreachable!(),
```

Comment:
sorry, I was thinking that `fn fips_mode()` didn't need to return an error, since I wasn't sure why it would fail. But now I see that `s2n_get_fips_mode` fails if s2n is not initialized, so there is a failure mode. And I guess `fn fips_mode()` should fail too if `s2n_get_fips_mode` started returning some new enum variant we didn't account for in the bindings yet. Anyway, I think this should probably go back to what you have before with `TryFrom`